# Entanglement
If you just using the basic superposition of quantum qubits you cannot efficiently get the result of calculation. Superposition is just a combination of bits and too many patterns exits. To filter the data and reduce the number of combination of result we can use entanglement.

You can syncronized more than 2qubits each other and filter the data just you want after the measurement.

## Circuit
Circuit is very simple, just apply CX gate after H gate.

In [ ]:
!pip install blueqat

In [2]:
from blueqat import Circuit

Circuit().h[0].cx[0,1].m[:].run(shots=100)

Counter({'00': 53, '11': 47})

Now you get almost half times of '00' and almost half times of '11'. Usually without the entanglement and just with superposition you will get result of 00,01,10,11. By using combination of H gate and CX gate you can filter the data you want efficiently.

# The entanglement of 3qubits / GHZ state
Greenberger–Horne–Zeilinger state is the state which as entanglement of 3qubits. Let's see the circuit.

In [1]:
from blueqat import Circuit                                                            

Circuit().h[:2].x[2].cx[1,2].cx[0,2].h[:].m[:].run(shots=100)

Counter({'000': 51, '111': 49})

We have half of '000' and another half of ''111'.